pkg/controller/backupbucket/actuator.go

```diff
@@ -36,18 +36,12 @@ func newActuator(mgr manager.Manager) backupbucket.Actuator {
 }
 
 func (a *actuator) Reconcile(ctx context.Context, _ logr.Logger, backupBucket *extensionsv1alpha1.BackupBucket) error {
-
 	backupConfig, err := helper.BackupConfigFromBackupBucket(backupBucket)
 	if err != nil {
 		return err
 	}
 
-	var cloudConfiguration *azure.CloudConfiguration
-	if backupBucket != nil {
-		cloudConfiguration = backupConfig.CloudConfiguration
-	}
-
-	azCloudConfiguration, err := azureclient.AzureCloudConfigurationFromCloudConfiguration(cloudConfiguration)
+	azCloudConfiguration, err := azureclient.AzureCloudConfigurationFromCloudConfiguration(backupConfig.CloudConfiguration)
```

Comment:
Is this really safe?
I mean backupConfig is still a pointer.
For this reason I would prefer that helpers like `BackupConfigFromBackupBucket` return a struct value instead of a pointer.